### DISCUSSION TITLE
Several Things! including mice,

### DISCUSSION THREAD

**Discussion by DoctorEam** [linuxmint/discussions#402] [Ideas]
Hi there!

Here's my most pressing ideas/observations:
(Updated with some solutions one can apply as a user).

- Configuring 2 mice individually (Possible to do in MX Linux). As an ambidextrous user, this is very necessary for correct mirroring of clicks and comfort (solvable with the xorg button remap solution).
- Adjusting scroll wheel speed (No distro has this 'feature' in it's settings!, IMO it's a fundamental thing to have). 
Solvable with the imwheel thing. Tbh I don't undestand people's obsession with DE's and their looks while often neglecting very basic functionality.
- The 'show desktop' button should stay on the left side of the bar.
- A minor issue is that Telegram has to be repinned to the taskbar, and the old launcher has to be removed after each update. Not sure if this happens with other programs as well. Definitely an issue with Telegram.
- The way MX linux is built is very good (except their horrible panel configurability), maybe Mint can adopt a few things from them. Just because of their unflexible panel I had to install cimmanon on MX.
- The startup applications menu in cinnamon is very practical, every distro and desktop environment should have this (as well as the mice flexibility). Most distros seem to have an autostart folder in .config where a .desktop file can be placed.

**Comment by 100savage** [comment]
This sounds like small amount of people would use something like this.
.001%

**Reply by DoctorEam** [reply]
Some of these things are basics, even in windows 95 (mouse scroll wheel speed buddy, c'mon). There are workarounds for some of them, which is good.
Copying the windows 'show desktop' button being on the right is not good. I bet many people use that button a lot, and like it to be on the left side.

**Comment by janos-r** [comment]
Being able to remap keys could be a good idea!
I just watched the youtube video from Dreams of Code "Turning the worst key on a keyboard into the most useful one" about remapping CapsLock. And that honestly sounds like an amazing idea. It would be great if the OS shipped with a GUI app for remapping keys.

**Reply by billyswong** [reply]
Try `input-remapper` for GUI key remapping.

**Comment by Jopp-gh** [comment]
>The 'show desktop' button should stay on the left side of the bar  

Right click on the panel, click on the toggle and edit the position of your applets via drag and drop. You can place applets wherever you want on the panel.

To remap buttons, press the super key and type `keyboard`. If you want something more, check out
http://superuser.com/questions/2194/ddg#5165

Well, my advice, try to use Mint for at least a month and you will think differently. If Mint is new to you, then your old habits may not work anymore or work differently as you would expect. BTW, check out Mintinstall, where you can download 3rd party apps to customize Mint.

On the forum of Linux Mint, you can ask for assistance to write your own shell scripts, if you need some features too.

**Comment by cplittlfield** [comment]
I agree that a scroll-speed setting for the mouse wheel would be useful.  The more options that are customizable, the better, in my view.  There is probably a way to set the scroll speed on the command line, but putting it in the GUI could be helpful.

I extremely left-handed, so I would never be using two mice.  In fact, I use a graphics pen and tablet instead of a mouse, and prefer it greatly (I do occasionally miss the mouse wheel, however :smile: )

I don't know what's going on with Telegram.  @DoctorEam, did you install it as a flatpak or as a snap?  Are you updating through the Update Manager (mintinstall) or downloading newer versions?  I wouldn't expect the problem you report if you are using the Update Manager.  That might be a bug.  However, the behavior you report is exactly what I would expect if you are downloading newer versions by hand.

A lot of desktop files get stored in /home/$USER/.local/share/applications and in /home/$USER/.config/autostart, not to mention various sub-directories of /usr/share/, in particular /usr/share/app-install/.

**Comment by billyswong** [comment]
For a standard computer mouse, a scroll wheel emits a scroll up or scroll down button event every time you feel the wheel gets a subtle bump within. `imwheel` hacks the scroll wheel speed by inserting extra number of wheel button events in between the real scroll up or scroll down events from your mouse or maybe block some of them out. It is a hack and may cause strange quirks for software that use the scroll wheel event in more cases than scrolling a continuous document.

For a true mouse scroll speed adjustment with zero unexpected side-effect, I can think of only 2 ways. 
1. Buy a more expensive mouse with high resolution scrolling technology, which will then be possible to emit more frequent or more sparse low resolution scroll event to legacy applications through driver setting. 
2. Get each individual GUI applications or the GUI toolkits underneath the applications be adjusted on how much should it scroll per mouse scroll event.

In other words, we need GTK, the GUI toolkit Linux Mint uses, to implement this feature in backend.

Edit: For Firefox, one can go to `about:config`, then experiment with all the "mousewheel" preferences there.
